VRCFaceTracking/Params/Expressions/Legacy/Lip/UnifiedSRanMapper.cs

```diff
@@ -0,0 +1,66 @@
+            { SRanipal_LipShape_v2.MouthUpperLeft, exp => exp.Shapes[(int)UnifiedExpressions.MouthUpperLeft].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthLowerRight, exp => exp.Shapes[(int)UnifiedExpressions.MouthLowerRight].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthLowerLeft, exp => exp.Shapes[(int)UnifiedExpressions.MouthLowerLeft].AdjustedWeight },
+
+            { SRanipal_LipShape_v2.MouthUpperOverturn, exp => (exp.Shapes[(int)UnifiedExpressions.LipFunnelUpperLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelUpperRight].AdjustedWeight) / 2.0f },
+            { SRanipal_LipShape_v2.MouthLowerOverturn, exp => (exp.Shapes[(int)UnifiedExpressions.LipFunnelLowerLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelLowerRight].AdjustedWeight) / 2.0f },
+
+            { SRanipal_LipShape_v2.MouthPout, exp => (exp.Shapes[(int)UnifiedExpressions.LipPuckerLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipPuckerRight].AdjustedWeight) / 2.0f },
+
+            { SRanipal_LipShape_v2.MouthSmileRight, exp => exp.Shapes[(int)UnifiedExpressions.MouthSmileRight].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthSmileLeft, exp => exp.Shapes[(int)UnifiedExpressions.MouthSmileLeft].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthSadRight, exp => Math.Min(1, exp.Shapes[(int)UnifiedExpressions.MouthFrownRight].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.MouthStretchRight].AdjustedWeight) },
+            { SRanipal_LipShape_v2.MouthSadLeft, exp => Math.Min(1, exp.Shapes[(int)UnifiedExpressions.MouthFrownLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.MouthStretchLeft].AdjustedWeight) },
+
+            { SRanipal_LipShape_v2.CheekPuffLeft, exp => exp.Shapes[(int)UnifiedExpressions.CheekPuffLeft].AdjustedWeight },
+            { SRanipal_LipShape_v2.CheekPuffRight, exp => exp.Shapes[(int)UnifiedExpressions.CheekPuffRight].AdjustedWeight },
+            { SRanipal_LipShape_v2.CheekSuck, exp => (exp.Shapes[(int)UnifiedExpressions.CheekSuckLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.CheekSuckRight].AdjustedWeight) / 2.0f },
+
+            { SRanipal_LipShape_v2.MouthUpperUpRight, exp => exp.Shapes[(int)UnifiedExpressions.MouthUpperUpRight].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelUpperRight].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthUpperUpLeft, exp => exp.Shapes[(int)UnifiedExpressions.MouthUpperUpLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelUpperLeft].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthLowerDownRight, exp => exp.Shapes[(int)UnifiedExpressions.MouthLowerDownRight].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelLowerRight].AdjustedWeight },
+            { SRanipal_LipShape_v2.MouthLowerDownLeft, exp => exp.Shapes[(int)UnifiedExpressions.MouthLowerDownLeft].AdjustedWeight + exp.Shapes[(int)UnifiedExpressions.LipFunnelLowerLeft].AdjustedWeight },
```

Comment:
What is the basis for using LipFunneler with MouthLowerDown and MouthUpperUp shapes?

Reply:
Lip Funneler opens the upper and lower lips, which Overturn does not have baked in. This will force the tracking interface to behave as SRanipal does with the respective SRanipal shape. The SRanipal module does the inverse of this operation and it seems to work as intended. I have yet to do a A/B test between old and new for this iteration but mathematically it should be able to be isolated properly.